### PR TITLE
Hamdan core and compose integrated for search pages

### PR DESCRIPTION
**Pull request by HamdanSCode**

### CODE REVIEW

File: searchCore/src/main/java/com/hamdan/forzenbook/search/core/viewmodel/BaseSearchResultViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+abstract class BaseSearchResultViewModel(
+    private val getPostByStringUseCase: GetPostByStringUseCase,
+    private val getPostByUserIdUseCase: GetPostByUserIdUseCase
+) : ViewModel() {
+
+    sealed interface SearchResultState {
+        val posts: List<PostData>
+
+        data class Content(override val posts: List<PostData> = listOf()) : SearchResultState
```
Comment (ForzenAcademy) [inline searchCore/src/main/java/com/hamdan/forzenbook/search/core/viewmodel/BaseSearchResultViewModel.kt:19]: Use `emptyList()` here for the empty lists you don't intend to use

File: searchCore/src/main/java/com/hamdan/forzenbook/search/core/viewmodel/BaseSearchViewModel.kt [hunk 2]
```diff
@@ -3,49 +3,68 @@ package com.hamdan.forzenbook.search.core.viewmodel
...
+    fun onUpdateSearch(text: String) {
+        updateSearchText(text)
+    }
+
+    fun navigateQuery(): String {
+        return "/-1/${searchState.query}/"
+    }
+
+    fun navigateUser(id: Int): String {
+        return "/$id/%20/"
```
Comment (ForzenAcademy) [inline searchCore/src/main/java/com/hamdan/forzenbook/search/core/viewmodel/BaseSearchViewModel.kt]: Constant, one line function

File: searchCore/src/main/java/com/hamdan/forzenbook/search/core/viewmodel/BaseSearchViewModel.kt [hunk 3]
```diff
@@ -3,49 +3,68 @@ package com.hamdan.forzenbook.search.core.viewmodel
...
+    fun onSearchSubmit(onSuccess: () -> Unit, onError: () -> Unit) {
+        searchByQuery(onSuccess, onError)
+    }
+
+    fun onUpdateSearch(text: String) {
+        updateSearchText(text)
+    }
+
+    fun navigateQuery(): String {
+        return "/-1/${searchState.query}/"
```
Comment (ForzenAcademy) [inline searchCore/src/main/java/com/hamdan/forzenbook/search/core/viewmodel/BaseSearchViewModel.kt]: Constant, one line function

**Review by ForzenAcademy — APPROVED** [review]
Small things

File: searchCore/src/main/java/com/hamdan/forzenbook/search/core/viewmodel/BaseSearchResultViewModel.kt (hunk 1 above)
Comment (HamdanSCode) [reply searchCore/src/main/java/com/hamdan/forzenbook/search/core/viewmodel/BaseSearchResultViewModel.kt:19]: roger

**Pull request merged**